### PR TITLE
Fix weekly emails

### PR DESCRIPTION
**Pull request by emilyb7**
I've made the following changes so far:
- change cron job to run for each site / location combination and generate an email for each
- adapt query logic so that we can pass in arguments (site and location); currently we are checking for SITE in the SQL query. We need to separately check for location.

Things to fix:
- all emails currently going to the goodmans fields address
- check for location where necessary and adapt email logic
- don't even want to look at the tests!
- change cron job timing back to once per week! (currently once per minute)

### CODE REVIEW

**Comment by sohilpandya** [comment]
Hey @emilyb7, can you please take a look at this, the tests are all passing and the emails are being sent with the correct site/location. 👍

File: src/server/model/email.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ const createSes = () => new aws.SES({ apiVersion: '2010-12-01' })
 
 const emailOpts = (subject, html, location, site) => {
   let emailExtension
-  if (emails.includes(location.toLowerCase())) {
+  if (location && emails.includes(location.toLowerCase())) {
     emailExtension = location.toLocaleLowerCase()
   } else if (emails.includes(site.toLocaleLowerCase())) {
     emailExtension = site.toLowerCase()
```
Comment (emilyb7) [inline src/server/model/email.js:23]: should we do `toLocaleLowerCase` here too? I can't remember why we did that anyway...

File: src/server/model/queries.js [hunk 2]
```diff
@@ -5,8 +5,17 @@ const init = (query, cb) => {
   runSqlFromFs(query, 'schema', {}, cb)
 }
 
-const weekly = (query, cb) => {
-  runSqlFromFs(query, 'weekly', {}, cb)
+// run query to get data for weekly email
+// args = [ site, location ]
+const weekly = (query, cb, args) => {
+  let [site, location] = args
+  if (location) {
```
Comment (emilyb7) [inline src/server/model/queries.js]: this is really nasty, can we change it?

Comment (emilyb7) [reply src/server/model/queries.js]: maybe: `location = location ? location : '' `

File: src/server/weekly_report.js [hunk 3]
```diff
@@ -29,24 +35,34 @@ const weeklyReport = (sendEmail, getReports) => {
...
+        if (err) {
+          console.log('email sending failed!')
+          return console.error(error)
+        }
 
-      console.log('summary email send successfully')
-    })
-  })
+        console.log('summary email sent successfully', location, site)
+        console.log('summary email sent successfully', response.rows)
```
Comment (emilyb7) [inline src/server/weekly_report.js]: let's get rid of this

File: src/server/start.js [hunk 4]
```diff
@@ -3,13 +3,34 @@ const cron = require('node-cron')
...
 
-    cron.schedule('59 22 * * 7', function () {
-      weeklyReport(server.plugins.model.sendEmail, server.plugins.model.weekly)
+server.start(startErr => {
+  if (startErr) throw startErr
+
+  server.plugins.model.init(modelErr => {
+    if (modelErr) throw modelErr
+
+    cron.schedule('* * * * *', () => {
```
Comment (emilyb7) [inline src/server/start.js:24]: still need to fix this!

Comment (emilyb7) [reply src/server/start.js:24]: can be in a separate PR, just don't want to forget...

File: src/server/model/queries.js [hunk 5]
```diff
@@ -5,8 +5,17 @@ const init = (query, cb) => {
...
+    location = location
+  } else {
+     location = ''
+  }
+  console.log(site, location, '<<<<<<<');
```
Comment (emilyb7) [inline src/server/model/queries.js]: get rid of this

File: test/server/model/helpers.js [hunk 6]
```diff
@@ -16,8 +16,9 @@ test('connect to database', t => {
 test('run some queries', t => {
   const query = connect(connectionConfig)
 
+console.log(query.toString());
```
Comment (emilyb7) [inline test/server/model/helpers.js]: !!

**Review by emilyb7 — COMMENTED** [review]
cool, few things we should tidy up though!

File: src/server/model/email.js (hunk 1 above)
Comment (sohilpandya) [reply src/server/model/email.js:23]: I don't think we need it here, we are just checking if it's undefined. Thoughts

**Comment by sohilpandya** [comment]
@emilyb7 Thanks for that, please merge in if the changes are ok, we won't be getting all OK from Travis due to coverage percentage. 

We need to update the chron, I'll create a seperate PR for that.

**Pull request merged**